Better handling of EK certificate processing errors

### Problem:
Paccor does not provide a reasonable message when it cannot read the EC.  

The stack trace below shows the error returned for the problem described in Issue #6.  The fix proposed in that issue would handle situations similar to the case we saw for that issue.  It could still not represent a valid EC.  

### Fix:
Better errors will help when paccor is not able to process the EC.  pc_certgen.sh should not reach the validation step if a platform certificate is not created.

### Stack trace:
```
Generating a signed Platform Credential
Exception in thread "main" org.bouncycastle.openssl.PEMException: problem parsing cert: java.io.IOException: Extra data detected in stream
        at org.bouncycastle.openssl.PEMParser$X509CertificateParser.parseObject(Unknown Source)
        at org.bouncycastle.openssl.PEMParser.readObject(Unknown Source)
        at cli.DeviceObserverCli.handleCommandLine(DeviceObserverCli.java:66)
        at cli.SigningCli.handleCommandLine(SigningCli.java:100)
        at cli.SigningCli.main(SigningCli.java:192)
Caused by: java.io.IOException: Extra data detected in stream
        at org.bouncycastle.asn1.ASN1Primitive.fromByteArray(Unknown Source)
        at org.bouncycastle.cert.CertUtils.parseNonEmptyASN1(Unknown Source)
        at org.bouncycastle.cert.X509CertificateHolder.parseBytes(Unknown Source)
        at org.bouncycastle.cert.X509CertificateHolder.<init>(Unknown Source)
        ... 5 more
Validating the signature
File "/opt/paccor/scripts/pc_testgen/platform_cert.20181005074720.crt" does not exist.
PC Credential Creation Complete.
Platform Credential has been placed in /opt/paccor/scripts/pc_testgen/platform_cert.20181005074720.crt
```

--- a/build.gradle
+++ b/build.gradle
@@ -115,7 +115,7 @@ ospackage {
     packageName='paccor'
     os=LINUX
     arch=NOARCH
-    version='1.0.4'
+    version='1.0.5'
     release='1'
 
     into '/opt/paccor'

--- a/scripts/pc_certgen.sh
+++ b/scripts/pc_certgen.sh
@@ -133,6 +133,10 @@ fi
 # Step 7 create and sign the new platform credential
 echo "Generating a signed Platform Credential"
 bash $signer_bin -x "$extsettings" -c "$componentlist" -e "$ekcert" -p "$policyreference" -k "$sigkey" -P "$pcsigncert" -N "$serialnumber" -b "$dateNotBefore" -a "$dateNotAfter" -f "$pccert" 
+if [ $? -ne 0 ]; then
+    echo "The signer could not produce a Platform Credential, exiting"
+    exit 1
+fi
 
 # Step 8 validate the signature
 echo "Validating the signature"

--- a/src/main/java/cli/DeviceObserverCli.java
+++ b/src/main/java/cli/DeviceObserverCli.java
@@ -63,8 +63,20 @@ public class DeviceObserverCli {
         Holder holder = null;
         InputStreamReader isr = new InputStreamReader(new ByteArrayInputStream(CliHelper.derToPem(ekCertFile, x509type.CERTIFICATE)));
         PEMParser p = new PEMParser(isr);
-        Object readObject = p.readObject();
-        p.close();
+        Object readObject = null;
+        try {
+            readObject = p.readObject();
+        } catch (IOException e) {
+            String msg = "The EK certificate could not be read.";
+            if (e.getMessage().contains("Extra data detected in stream")) {
+                msg += " Check for extra bytes in the given file.";
+            }
+            throw new IOException(msg, e);
+        } finally {
+            if (p != null) {
+                p.close();
+            }
+        }
         if (readObject instanceof X509CertificateHolder) {
             ekCert = (X509CertificateHolder)readObject;
             holder = new Holder(new IssuerSerial(ekCert.getIssuer(), ekCert.getSerialNumber()));

